app/lib/transformNpmCommand.ts

```diff
@@ -0,0 +1,21 @@
+export type PackageManager = "npm" | "yarn" | "pnpm" | "bun";
+
+export function transformNpmCommand(
+  prefix: string,
+  cmd: string,
+  packageManagerTarget: "yarn" | "bun" | "pnpm" | "npm",
+) {
+  if (prefix === "npm") {
+    if (cmd.split(" ")[0] === "install" && packageManagerTarget === "yarn") {
+      return `${packageManagerTarget} ${cmd.replace("install", "add")}`;
+    }
+    return `${packageManagerTarget} ${cmd}`;
+  }
+  switch (packageManagerTarget) {
+    case "bun":
+      return `bunx ${cmd}`;
+    case "pnpm":
+      return `pnpm dlx ${cmd}`;
+  }
+  return `${prefix} ${cmd}`;
+}
```

Comment:
I know we don't have a ton of tests in this repo, but this seems like a really good place for a unit test. Would you mind adding one with various use cases?

Also, small nit, will you change this file name to `transform-npm-command.ts`?